### PR TITLE
Feature/477 job application

### PR DESCRIPTION
**Pull request by BlancoSebastianEzequiel**

### CODE REVIEW

**Review by dylanalvarez — APPROVED** [review]

File: src/models/JobApplication/Errors/AdminCannotModerateJobApplicationError.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { Admin } from "$models";
+
+export class AdminCannotModerateJobApplicationError extends Error {
+  public static buildMessage({ secretary }: Admin) {
+    return `admin from ${secretary} secretary cannot moderate jobApplication`;
```
Comment (llmanuel) [inline src/models/JobApplication/Errors/AdminCannotModerateJobApplicationError.ts:5]: can't or can not

File: src/models/Permissions/UserPermissions/Permissions.ts [hunk 2]
```diff
@@ -18,6 +18,10 @@ export class UserPermissions implements IPermissions {
     return this.hasPermission(permission => permission.canModerateOffer(offer));
   }
 
+  public async canModerateJobApplication(jobApplication: JobApplication) {
```
Comment (llmanuel) [inline src/models/Permissions/UserPermissions/Permissions.ts]: no async

File: test/models/Permissions/AdminPermissions/ApprovedApplicantAdminPermission.test.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+import { UUID } from "$models/UUID";
+
+describe("ApprovedApplicantAdminPermission", () => {
+  let applicant: Applicant;
+
+  beforeAll(() => {
+    applicant = new Applicant({ padron: 1, userUuid: UUID.generate() });
+  });
+
+  it("returns true the applicant is approved", async () => {
```
Comment (llmanuel) [inline test/models/Permissions/AdminPermissions/ApprovedApplicantAdminPermission.test.ts]: returns true if the applicant is approved

File: test/models/Permissions/AdminPermissions/ApprovedApplicantAdminPermission.test.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    applicant.set({ approvalStatus: ApprovalStatus.approved });
+    const permission = new ApprovedApplicantAdminPermission(applicant);
+    expect(await permission.apply()).toBe(true);
+  });
+
+  it("returns false the applicant is rejected", async () => {
```
Comment (llmanuel) [inline test/models/Permissions/AdminPermissions/ApprovedApplicantAdminPermission.test.ts]: if

File: test/models/Permissions/AdminPermissions/ApprovedApplicantAdminPermission.test.ts [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+    applicant.set({ approvalStatus: ApprovalStatus.rejected });
+    const permission = new ApprovedApplicantAdminPermission(applicant);
+    expect(await permission.apply()).toBe(false);
+  });
+
+  it("returns false the applicant is pending", async () => {
```
Comment (llmanuel) [inline test/models/Permissions/AdminPermissions/ApprovedApplicantAdminPermission.test.ts]: if

**Review by llmanuel — COMMENTED** [review]
mini comments

**Pull request merged**